src/main/java/com/denizenscript/clientizen/events/ScreenOpenCloseEvent.java

```diff
@@ -17,49 +27,163 @@ public class ScreenOpenCloseEvent extends ScriptEvent {
+        registerScreenName(SelectWorldScreen.class, "select_world");
+        registerScreenName(SymlinkWarningScreen.class, "symlink_warning");
+        // Other screens
+        registerScreenName(AccessibilityOnboardingScreen.class, "accessibility_onboarding");
+        registerScreenName(AddServerScreen.class, "add_server");
+        registerScreenName(BackupPromptScreen.class, "backup_prompt");
+        registerScreenName(ChatScreen.class, "chat");
+        registerScreenName(ConfirmLinkScreen.class, "confirm_link");
+        registerScreenName(ConfirmScreen.class, "confirm");
+        registerScreenName(ConnectScreen.class, "connect");
+        registerScreenName(CreditsScreen.class, "credits");
+        registerScreenName(CustomizeBuffetLevelScreen.class, "customize_buffet_level");
+        registerScreenName(CustomizeFlatLevelScreen.class, "customize_flat_level");
+        registerScreenName(DatapackFailureScreen.class, "datapack_failure");
+        registerScreenName(DeathScreen.class, "death");
+        registerScreenName(DemoScreen.class, "demo");
+        registerScreenName(DialogScreen.class, "dialog");
+        registerScreenName(DirectConnectScreen.class, "direct_connect");
+        registerScreenName(DisconnectedScreen.class, "disconnected");
+        registerScreenName(DownloadingTerrainScreen.class, "downloading_terrain");
+        registerScreenName(FatalErrorScreen.class, "fatal_error");
+        registerScreenName(GameMenuScreen.class, "game_menu");
+        registerScreenName(GameModeSelectionScreen.class, "game_mode_selection");
+        registerScreenName(LevelLoadingScreen.class, "level_loading");
+        registerScreenName(MessageScreen.class, "message");
+        registerScreenName(NoticeScreen.class, "notice_screen");
+        registerScreenName(OpenToLanScreen.class, "open_to_lan");
+        registerScreenName(OutOfMemoryScreen.class, "out_of_memory");
+        registerScreenName(PresetsScreen.class, "presets");
+        registerScreenName(ProgressScreen.class, "progress");
+        registerScreenName(Realms32BitWarningScreen.class, "realms_32_bit_warning");
+        registerScreenName(SleepingChatScreen.class, "sleeping_chat");
+        registerScreenName(StatsScreen.class, "stats");
+        registerScreenName(TaskScreen.class, "task");
+        registerScreenName(TitleScreen.class, "title");
+    }
+
+    private static final Map<Class<? extends Screen>, String> TYPE_MAP;
+
+    public static void registerScreenName(Class<? extends Screen> screenType, String name) {
+        TYPE_MAP.put(screenType, name);
+    }
+
+    public static String getScreenName(Class<? extends Screen> screenType) {
+        return TYPE_MAP.computeIfAbsent(screenType, clazz -> {
+            String className = clazz.getSimpleName();
+            if (className.endsWith("Screen")) {
+                className = className.substring(0, className.length() - "Screen".length());
+            }
```

Comment:
This seems like it won't work in a production environment, unless you're specifically targeting modded ones (in this case, include a comment about it)

Reply:
Added a comment